web/src/components/itemList.js

```diff
@@ -113,15 +97,17 @@ class ItemList extends Component {
                        onCompleted={data => {
                            const selected = [];
                            const currency = [];
+                           
                            for(let i = 0; i < data.getItems.length; i++) {
-                               selected.push(0);
+                               selected.push(data.getItems[i].select);
                                currency.push(false);
                            }
-                           this.selected = selected;
+
+                           this.props.initSelected(selected);
+                    
                            this.setState({
                                items: data.getItems,
                                currency: currency,
```

Comment:
Since `ShoppingBasket` also managing these data, why are you managing these here?